core/routes.go

```diff
@@ -26,65 +28,206 @@ func updateRoutes(pg *postgres) {
+
+	query := `
+		SELECT id, flight, last_seen_lat, last_seen_lon
+		FROM aircraft_data
+		WHERE 
+			hex != '' AND
+			flight != '' AND
+			route_processed = false
+		ORDER BY first_seen ASC`
+
+	rows, err := pg.db.Query(context.Background(), query)
+
+	if err != nil {
+		fmt.Println("unprocessedRoutes() - Error querying db: ", err)
+		return nil
+	}
+	defer rows.Close()
+
+	var aircrafts []Aircraft
+
+	for rows.Next() {
 
-	for _, aircraft := range new {
+		var aircraft Aircraft
 
-		route, err := getRoute(aircraft)
+		err := rows.Scan(
+			&aircraft.Id,
+			&aircraft.Flight,
+			&aircraft.LastSeenLat,
+			&aircraft.LastSeenLon,
+		)
 
 		if err != nil {
-			fmt.Println("Error getting route: ", err)
-			continue
+			fmt.Println("unprocessedRoutes() - Error scanning rows: ", err)
+			return nil
 		}
 
-		if route.Response.Flightroute.Callsign == "" {
-			existing = append(existing, aircraft)
+		aircrafts = append(aircrafts, aircraft)
+	}
+
+	fmt.Println("Aircrafts that have not have routes processed: ", len(aircrafts))
+	return aircrafts
+}
+
+func checkRouteExists(pg *postgres, aircraftToProcess []Aircraft) (existing []Aircraft, new []Aircraft) {
+
+	var callsignValues []string
+	for _, a := range aircraftToProcess {
+		callsignValues = append(callsignValues, a.Flight)
+	}
+
+	existingRoutes := make(map[string]*Aircraft)
+
+	query := `
+		SELECT id, route_callsign
+		FROM route_data
+		WHERE route_callsign = ANY($1::text[])
+		  AND last_updated IS NOT NULL
+		  AND last_updated > NOW() - INTERVAL '1 hour'`
+
+	rows, err := pg.db.Query(context.Background(), query, callsignValues)
+
+	if err != nil {
+		fmt.Println("checkRouteExists() - Error querying db: ", err)
+		return nil, nil
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var route Aircraft
+		err := rows.Scan(
+			&route.Id,
+			&route.Flight,
+		)
+
+		if err != nil {
+			fmt.Println("checkRouteExists() - Error scanning rows: ", err)
 			continue
 		}
 
-		routes = append(routes, *route)
-		existing = append(existing, aircraft)
+		existingRoutes[route.Flight] = &route
+	}
 
+	for _, a := range aircraftToProcess {
+		if _, ok := existingRoutes[a.Flight]; ok {
+			existing = append(existing, a)
+		} else {
+			new = append(new, a)
+		}
 	}
 
-	insertRoutes(pg, routes)
+	return existing, new
 
-	MarkProcessed(pg, "route_processed", existing)
+}
+
+func getRoutes(aircrafts []Aircraft) ([]RouteInfo, error) {
+
+	requestBodyData := buildRouteApiRequestBody(aircrafts)
+	requestBodyJson, err := json.Marshal(requestBodyData)
+
+	if err != nil {
+		return nil, err
+	}
+
+	url := "https://adsb.im/api/0/routeset"
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(requestBodyJson))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", fmt.Sprintf("Skystats/%s", version))
+	req.Header.Set("Accept", "application/json")
+
+	client := &http.Client{
+		Timeout: 10 * time.Second,
```

Comment:
API response time for up to 100 routes tends to be under 100ms (except when I'm restarting the service or something else is down). Add SSL handshake time to that, it still should stay below 1s

Reply:
Have reduced to 5s - still gives generous room for any network oddities.